packfile.c

```diff
@@ -309,7 +309,7 @@ void close_pack_windows(struct packed_git *p)
 	}
 }
 
-static int close_pack_fd(struct packed_git *p)
+int close_pack_fd(struct packed_git *p)
```

Comment:
This wouldn't be needed without the optimization above but I don't see any problem making this public.

Reply:
right. the new verify loop completely verifies all objects in one packfile before moving to the next packfile (because of the sort).  But when we hit 2000+ packfiles in the directory, visiting the next packfile requires us to free up a fd, and
this triggers the LRU search in close_one_pack().   So by closing the previous packfile in that loop, we'll only have 1
packfile open and avoid all of the LRU searching (which is O(n^2)).